### PR TITLE
Fixes issue #3353

### LINKED ISSUE #3353: Translation problems between language and source code language
**Describe the bug**
The problem is that there are translation differences between `ticker` and `tickerSymbol` both in the *.properties and in the source code. There is no uniform structure.

In German, the security ticker is named "Symbol". 
In English, this is called "ticker".

**Expected behavior**
The source code should be kept in English.

In order to create a uniformity, the variables, both in the source code and in the *.properties are to be corrected.
As more german developers are working on the project, I would suggest to rename the variables to `tickerSymbol`.

Then the translations are to be corrected.
As an example... the German translation in "Symbol" and not "Ticker Symbol, "Ticker-Symbol", "Tickersymbol" or "Ticker".
The same applies to the other available languages.

### PR DESCRIPTION
**Pull request by Nirus2000**
Fixes #3353 

#### German
   - https://de.wikipedia.org/wiki/Tickersymbol --> "Symbol"

#### Danish
   - https://da.wikipedia.org/wiki/Tickersymbol --> "Tickersymbol" --> "Symbol"
   - In one sentence --> "tickersymbol " or "aktiesymbol"

#### French
   - https://fr.wikipedia.org/wiki/Code_mn%C3%A9monique --> "Mnémo"
   - "Mnémo" is the abbreviation of "code mnémonique"
   - In one sentence --> "code mnémonique"

#### Italian
https://www.borsaitaliana.it/borsa/azioni/global-equity-market/dati-completi.html?isin=US88160R1014&lang=it
   - See "Dati Completi" --> "Codice Alfanumerico" --> lets get "Simbolo"
   - In one sentence --> "Codice Alfanumerico" or "Simbolo di borsa"

#### Dutch
   - https://nl.wikipedia.org/wiki/Ticker_symbol --> "Symbool"
   - In one sentence --> "ticker symbol"
   - "Ticker" is wrong!!
      - The term "ticker" in Dutch refers to the narrow strip of paper printed by a telegraph at the time that produced a ticking sound.

#### Polish
   - https://pl.wikipedia.org/wiki/Symbol_gie%C5%82dowy --> "Symbol giełdowy" --> "Symbol"
   - In one sentence --> "Symbol giełdowy"

#### Portuguese
   - https://br.advfn.com/bolsa-de-valores/nasdaq/TSLA/cotacao --> "Código da Ação" --> "Símbolo"
   - In one sentence --> "Código da Ação"

#### Russian
   - https://ru.wikipedia.org/wiki/%D0%A2%D0%B8%D0%BA%D0%B5%D1%80 --> "Ти́кер"

#### Slovak
   - "Symbol"
   - In one sentence --> "Akciový symbol"
   - Financial lexicon with technical terms --> https://iad.sk/slovnik-investora/

#### Spanish
   - https://es.wikipedia.org/wiki/Etiqueta_de_cotizaci%C3%B3n --> "Ticker"
   - Example: https://www.bolsasymercados.es/bme-exchange/es/Mercados-y-Cotizaciones/Acciones/Mercado-Continuo/Ficha/Promotora-De-Informac-Clase-A-Cancelado-ES0171743117
   - In one sentence --> "etiqueta de cotización" or "código de cotización"

#### Czech
   - https://cs.wikipedia.org/wiki/Ticker --> "Ticker"
   - Example: https://www.pse.cz/detail/DE0005493092

#### English
   - https://en.wikipedia.org/wiki/Ticker_symbol --> "Ticker"
   - In one sentence --> "ticker symbol" or "stock symbol"

### CODE REVIEW

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVPortfolioExtractor.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
         fields.add(new Field("time", Messages.CSVColumn_Time).setOptional(true)); //$NON-NLS-1$
 
         fields.add(new ISINField("isin", Messages.CSVColumn_ISIN).setOptional(true)); //$NON-NLS-1$
-        fields.add(new Field("ticker", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
```
Comment (buchen) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVPortfolioExtractor.java:38]: This is a breaking change for existing users. The "code" used in the JSON configurations that the user can create (and are stored locally). I do not think users are editing or seeing the JSON, therefore I tend to say: let's not migrate the configurations but rather leave the code "ticker" here.

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVPortfolioTransactionExtractor.java [hunk 2]
```diff
@@ -41,7 +41,7 @@
         fields.add(new DateField("date", Messages.CSVColumn_Date)); //$NON-NLS-1$
         fields.add(new Field("time", Messages.CSVColumn_Time).setOptional(true)); //$NON-NLS-1$
         fields.add(new ISINField("isin", Messages.CSVColumn_ISIN).setOptional(true)); //$NON-NLS-1$
-        fields.add(new Field("ticker", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
+        fields.add(new Field("tickerSymbol", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
```
Comment (buchen) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVPortfolioTransactionExtractor.java:44]: dito

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVSecurityExtractor.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
         List<Field> fields = getFields();
         fields.add(new Field("isin", Messages.CSVColumn_ISIN).setOptional(true)); //$NON-NLS-1$
         fields.add(new Field("wkn", Messages.CSVColumn_WKN).setOptional(true)); //$NON-NLS-1$
-        fields.add(new Field("ticker", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
+        fields.add(new Field("tickerSymbol", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
```
Comment (buchen) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVSecurityExtractor.java:28]: dito

File: name.abuchen.portfolio/src/name/abuchen/portfolio/json/JSecurity.java [hunk 4]
```diff
@@ -9,7 +9,7 @@
     private String name;
     private String isin;
     private String wkn;
-    private String ticker;
+    private String tickerSymbol;
     private String currency;
```
Comment (buchen) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/json/JSecurity.java:13]: This is a breaking change in the JSON that the user can download the view "All transactions" -> Toolbar: Export -> All transactions (JSON). Why? The json is created using the property names of the Java class. 

To be honest, I do not know how many users are using this export. So maybe we can get away with the change here

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVAccountTransactionExtractor.java [hunk 5]
```diff
@@ -38,7 +38,7 @@
         fields.add(new DateField("date", Messages.CSVColumn_Date)); //$NON-NLS-1$
         fields.add(new Field("time", Messages.CSVColumn_Time).setOptional(true)); //$NON-NLS-1$
         fields.add(new ISINField("isin", Messages.CSVColumn_ISIN).setOptional(true)); //$NON-NLS-1$
-        fields.add(new Field("ticker", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
+        fields.add(new Field("tickerSymbol", Messages.CSVColumn_TickerSymbol).setOptional(true)); //$NON-NLS-1$
```
Comment (buchen) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVAccountTransactionExtractor.java:41]: also a breaking change

**Comment by buchen** [comment]
I also did some research. I think the term "ticker" is not really used anymore on English/American sites. I do not find it on Yahoo nor on Google. Many others do not have that term on the site at all - neither symbol nor ticker - they just show the symbol.

Example Yahoo:
Bildschirmfoto 2023-05-19 um 21 44 21

Example Google:
Bildschirmfoto 2023-05-19 um 21 48 12


I like your proposal:
* Use internally always "tickerSymbol". There are thousands of usages. I do not see any benefit of mangling up all file histories just for this change.
* In external (new!) API, we use "symbol"
* In the translation we use the locally relevant terms. I assume this is mostly "symbol"
* The changes to the CSV importer we should revert - it is internally and not visible - no need to break existing users

I'll not be able to adapt the change now (Friday) but hope to be able to do it tomorrow

**Comment by Pnda87** [comment]
> I also did some research. I think the term "ticker" is not really used anymore on English/American sites. I do not find it on Yahoo nor on Google. Many others do not have that term on the site at all - neither symbol nor ticker - they just show the symbol.

I disagree.

Bloomberg, the global leader in financial and market data uses the term 'ticker' to identify financial instruments and indices. Same goes for MSCI.
Refinitiv (Reuters) uses the term 'symbol' or 'RIC'.

However, in the day to day business in the financial industry in German-speaking countries, the term "ticker" is predominantly used.

**Comment by Nirus2000** [comment]
Okay, that's very interesting.
Overall, we should find a solution for this, which term we use globally in the source code and which in the translations. (highest common denominator)

**Pull request closed without merging**